pkg/webhook/webhook.go

```diff
@@ -325,19 +328,22 @@ func getINFList(ctx context.Context, kubeClient client.Client) (*ingressnodefwv1
 }
 
 func validateAgainstExistingINFs(allErrs field.ErrorList, infList *ingressnodefwv1alpha1.IngressNodeFirewallList, newSourceCIDRs []string,
-	newRules []ingressnodefwv1alpha1.IngressNodeFirewallProtocolRule, newINFRulesIndex int, newINFName string) field.ErrorList {
+	newRules []ingressnodefwv1alpha1.IngressNodeFirewallProtocolRule, newINFRulesIndex int, newINFName string, newNodeSelector v1.LabelSelector) field.ErrorList {
 
 	for _, existingINF := range infList.Items {
 		existingINFName := existingINF.Name
-		for _, existingRules := range existingINF.Spec.Ingress {
-			for _, existingSourceCIDR := range existingRules.SourceCIDRs {
-				for _, newSourceCIDR := range newSourceCIDRs {
-					if strings.TrimSpace(newSourceCIDR) == strings.TrimSpace(existingSourceCIDR) {
-						if existingINFName != newINFName && isOrderOverlapping(existingRules.FirewallProtocolRules, newRules) {
-							allErrs = append(allErrs,
-								field.Invalid(field.NewPath("spec").Child("ingress").Index(newINFRulesIndex).Key("rules"),
-									newINFName, fmt.Sprintf("order is not unique for sourceCIDR %q and conflicts with "+
-										"IngressNodeFirewall %q", newSourceCIDR, existingINF.Name)))
+		// Need to validate rules only if they are applied to the same Nodes
+		if reflect.DeepEqual(existingINF.Spec.NodeSelector, newNodeSelector) {
```

Comment:
Curious, did you look for any tools to compare labels instead of using reflect?
This nested loops made my head spin :)
Needs factoring but doesnt need to be here.
/lgtm